### PR TITLE
OHRM5X-1707: Fix Issues In Translation

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: installer/Migration/V5_2_0/lang-string/pim.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+langStrings:
+  -
+    value: 'State Income Tax'
+    unitId: state_income_tax
```
Comment (Super-Chama) [inline installer/Migration/V5_2_0/lang-string/pim.yaml:4]: Need line ending. also check if pim module is enabled for migration

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue [hunk 2]
```diff
@@ -95,13 +95,16 @@ import {freshDate, formatDate} from '@ohrm/core/util/helper/datefns';
...
+
+const {$t} = usei18n();
 
 const attendanceRecordNormalizer = data => {
   return data.map(item => {
     return {
       id: item.empNumber,
       empName: `${item?.firstName} ${item?.lastName}
-          ${item?.terminationId ? ' (Past Employee)' : ''}`,
+          ${item?.terminationId ? ` ${$t('general.past_employee')}` : ''}`,
```
Comment (Super-Chama) [inline src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:107]: Composable functions should be exclusively used within the setup context. 
https://vuejs.org/guide/reusability/composables.html#using-composables-in-options-api

```
<template>
  <div>test</div>
</template>

<script>
// outside of component context
// no access to any reactive variables/methods
// only used for imports and static functions

export default {
  name: 'SomeComponent',
  setup() {
    // the setup context
    // this is called first, hence it don't have access to whats defined in data()
    // all the composables are consumed here
    // dont have access to mixins or global attributes
    // unless explicitly defined
  },
  data() {
    // component context
    // has access to whats defined in setup
    // if returned from setup
  },
  // or computed/ other hooks
  methods: {
    // component context
    // has access to whats defined in setup
    // if returned from setup
  }
};
</script>
```

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue [hunk 3]
```diff
@@ -260,7 +262,9 @@ export default {
       if (employee) {
         const name = `${employee.firstName} ${employee.middleName}
         ${employee.lastName}`;
-        return `${name} ${employee.terminationId ? '(Past Employee)' : ''}`;
+        return `${name} ${
```
Comment (Super-Chama) [inline src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:265]: For Transforming employee object to full name/display name with past employee tag use [useEmployeeNameTranslate](https://github.com/orangehrm/orangehrm/blob/main/src/client/src/core/util/composable/useEmployeeNameTranslate.ts)

```
...
setup() {
    const {$tEmpName} = useEmployeeNameTranslate();

    return {
       // vue don't allow variables prefixed with $ sign to return from setup
      translateEmpName: $tEmpName,
    };
},

computed: {
    employeeName() {
      const employee = this.response?.meta?.employee;
      if (employee) {
        return this.translateEmpName(employee, {
            includeMiddle: true,
            excludePastEmpTag: false,
        });
      }
      return '';
    }
}
```

[ref](https://github.com/orangehrm/orangehrm/blob/faf46890b5b1aa56bf1756d9fbea2118e372b971/src/client/src/orangehrmRecruitmentPlugin/pages/ViewActionHistory.vue#L256)

File: src/client/src/orangehrmPimPlugin/pages/employee/Employee.vue [hunk 4]
```diff
@@ -214,7 +209,14 @@ export default {
       });
     };
 
-    const filters = ref({...defaultFilters});
+    const filters = ref({
```
Comment (Super-Chama) [inline src/client/src/orangehrmPimPlugin/pages/employee/Employee.vue:212]: seems default filters are not re-used. you can even directly define here

File: src/client/src/orangehrmTimePlugin/pages/project/Project.vue [hunk 5]
```diff
@@ -117,6 +118,8 @@ import ProjectAutocomplete from '@/orangehrmTimePlugin/components/ProjectAutocom
 import CustomerAutocomplete from '@/orangehrmTimePlugin/components/CustomerAutocomplete.vue';
 import ProjectAdminAutocomplete from '@/orangehrmTimePlugin/components/ProjectAdminAutocomplete.vue';
 
+const {$t} = usei18n();
```
Comment (Super-Chama) [inline src/client/src/orangehrmTimePlugin/pages/project/Project.vue:121]: define this inside setup

File: src/client/src/orangehrmPimPlugin/pages/employee/EmployeeJob.vue [hunk 6]
```diff
@@ -338,7 +341,7 @@ export default {
         return {
           id: jobTitle.id,
           label: jobTitle?.deleted
-            ? jobTitle.label + ' (Deleted)'
+            ? jobTitle.label + `${$t('general.deleted')}`
```
Comment (Super-Chama) [inline src/client/src/orangehrmPimPlugin/pages/employee/EmployeeJob.vue:344]:
```suggestion
            ? jobTitle.label + ` ${this.$t('general.deleted')}`
```
inside the component context you can access to global method $t (via mixin), no need to use `usei18n`

File: src/client/src/orangehrmPimPlugin/pages/employee/EmployeeJob.vue [hunk 7]
```diff
@@ -204,6 +204,9 @@ import {
 import useDateFormat from '@/core/util/composable/useDateFormat';
 import {formatDate, parseDate} from '@/core/util/helper/datefns';
 import useLocale from '@/core/util/composable/useLocale';
+import usei18n from '@/core/util/composable/usei18n';
+
+const {$t} = usei18n();
```
Comment (Super-Chama) [inline src/client/src/orangehrmPimPlugin/pages/employee/EmployeeJob.vue:209]: remove. see below comment

File: src/client/src/orangehrmTimePlugin/pages/timesheets/EditTimeSheet.vue [hunk 8]
```diff
@@ -236,7 +239,9 @@ export default {
         return this.$t('time.edit_timesheet');
       } else if (this.employee) {
         const empName = this.employee?.terminationId
-          ? `${this.employee.firstName} ${this.employee.lastName} (Past Employee)`
+          ? `${this.employee.firstName} ${this.employee.lastName} ${$t(
```
Comment (Super-Chama) [inline src/client/src/orangehrmTimePlugin/pages/timesheets/EditTimeSheet.vue:242]: see above comments. use [useEmployeeNameTranslate](https://github.com/orangehrm/orangehrm/blob/main/src/client/src/core/util/composable/useEmployeeNameTranslate.ts)

File: src/client/src/orangehrmTimePlugin/pages/timesheets/ViewEmployeeTimesheet.vue [hunk 9]
```diff
@@ -174,7 +177,9 @@ export default {
   computed: {
     title() {
       const empName = this.employee.terminationId
-        ? `${this.employee.firstName} ${this.employee.lastName} (Past Employee)`
+        ? `${this.employee.firstName} ${this.employee.lastName}  ${$t(
```
Comment (Super-Chama) [inline src/client/src/orangehrmTimePlugin/pages/timesheets/ViewEmployeeTimesheet.vue:180]: See above comments use [useEmployeeNameTranslate](https://github.com/orangehrm/orangehrm/blob/main/src/client/src/core/util/composable/useEmployeeNameTranslate.ts)

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue [hunk 10]
```diff
@@ -138,6 +129,17 @@ export default {
       };
     });
 
+    const attendanceRecordNormalizer = data => {
+      return data.map(item => {
+        return {
+          id: item.empNumber,
+          empName: `${item?.firstName} ${item?.lastName}
+          ${item?.terminationId ? $t('general.past_employee') : ''}`,
```
Comment (RajithaKumara) [inline src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:137]:
```suggestion
          empName: this.translateEmpName(employee, { includeMiddle: false, excludePastEmpTag: false, });,
```

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue [hunk 11]
```diff
@@ -120,6 +110,7 @@ export default {
   },
 
   setup(props) {
+    const {$t} = usei18n();
```
Comment (RajithaKumara) [inline src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:113]:
```suggestion
    const {$tEmpName} = useEmployeeNameTranslate();
```

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue [hunk 12]
```diff
@@ -137,9 +137,10 @@ export default {
     );
 
     const {leaveActions, processLeaveAction} = useLeaveActions(http);
-    const {$t} = usei18n();
+
     const {jsDateFormat} = useDateFormat();
     const {locale} = useLocale();
+    const {$t} = usei18n();
```
Comment (RajithaKumara) [inline src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:143]: revert this change

File: src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue (hunk 3 above)
Comment (RajithaKumara) [reply src/client/src/orangehrmLeavePlugin/pages/LeaveRequest.vue:264]: seems not done with `useEmployeeNameTranslate`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmAdminPlugin/Api/I18NGroupAPI.php [hunk 13]
```diff
@@ -40,6 +40,7 @@ class I18NGroupAPI extends Endpoint implements CollectionEndpoint
     public function getAll(): EndpointResult
     {
         $i18NGroupSearchFilterParams = new I18NGroupSearchFilterParams();
+        $i18NGroupSearchFilterParams->setSortField(null);
```
Comment (RajithaKumara) [inline src/plugins/orangehrmAdminPlugin/Api/I18NGroupAPI.php:43]: *

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by RajithaKumara — APPROVED** [review]

File: src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue [hunk 14]
```diff
@@ -95,17 +95,7 @@ import {freshDate, formatDate} from '@ohrm/core/util/helper/datefns';
 import EmployeeAutocomplete from '@/core/components/inputs/EmployeeAutocomplete';
 import {yearRange} from '@/core/util/helper/year-range';
 import useDateFormat from '@/core/util/composable/useDateFormat';
-
-const attendanceRecordNormalizer = data => {
-  return data.map(item => {
-    return {
-      id: item.empNumber,
-      empName: `${item?.firstName} ${item?.lastName}
```
Comment (Super-Chama) [inline src/client/src/orangehrmAttendancePlugin/pages/ViewEmployeeAttendanceSummary.vue:103]: @ManulMax check here where the employee is coming from
